core/src/main/scala/cats/std/function.scala

```diff
@@ -71,13 +71,41 @@ trait Function1Instances {
       def compose[A, B, C](f: B => C, g: A => B): A => C = f.compose(g)
     }
 
-  implicit def function1Monoid[A,B](implicit B: Monoid[B]): Monoid[A => B] =
-    new Monoid[A => B] {
-      def empty: A => B = _ => B.empty
-      def combine(x: A => B, y: A => B): A => B = { a =>
-        B.combine(x(a), y(a))
-      }
-    }
+  implicit def function1Monoid[A,B](implicit M: Monoid[B]): Monoid[A => B] =
+    new Function1Monoid[A, B] { def B: Monoid[B] = M }
+
+  implicit val function1MonoidK: MonoidK[Lambda[A => A => A]] =
+    new Function1MonoidK {}
+}
+
+trait Function1Instances0 {
```

Comment:
Could you make this `private[std] sealed trait Function1Instances0` please? This will make it consistent with the work done in #612.